generation/templates/auth/OAuthAuthenticator.mustache

```diff
@@ -0,0 +1,97 @@
+{{>partial_header}}
+
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace {{packageName}}.Client.Auth
+{
+    /// <summary>
+    /// An authenticator for OAuth2 authentication flows
+    /// </summary>
+    public class OAuthAuthenticator : AuthenticatorBase
+    {
+        readonly string _tokenUrl;
+        readonly string _clientId;
+        readonly string _clientSecret;
+        readonly string _grantType;
+        readonly JsonSerializerSettings _serializerSettings;
+        readonly IReadableConfiguration _configuration;
+
+        /// <summary>
+        /// Initialize the OAuth2 Authenticator
+        /// </summary>
+        public OAuthAuthenticator(
+            string tokenUrl,
+            string clientId,
+            string clientSecret,
+            OAuthFlow? flow,
+            JsonSerializerSettings serializerSettings,
+            IReadableConfiguration configuration) : base("")
+        {
+            _tokenUrl = tokenUrl;
+            _clientId = clientId;
+            _clientSecret = clientSecret;
+            _serializerSettings = serializerSettings;
+            _configuration = configuration;
+
+            switch (flow)
+            {
+                /*case OAuthFlow.ACCESS_CODE:
+                    _grantType = "authorization_code";
+                    break;
+                case OAuthFlow.IMPLICIT:
+                    _grantType = "implicit";
+                    break;
+                case OAuthFlow.PASSWORD:
+                    _grantType = "password";
+                    break;*/
+                case OAuthFlow.APPLICATION:
+                    _grantType = "client_credentials";
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Creates an authentication parameter from an access token.
+        /// </summary>
+        /// <param name="accessToken">Access token to create a parameter from.</param>
+        /// <returns>An authentication parameter.</returns>
+        protected override async ValueTask<Parameter> GetAuthenticationParameter(string accessToken)
+        {
+            var token = string.IsNullOrEmpty(Token) ? await GetToken().ConfigureAwait(false) : Token;
+            return new HeaderParameter(KnownHeaders.Authorization, token);
+        }
+
+        /// <summary>
+        /// Gets the token from the OAuth2 server.
+        /// </summary>
+        /// <returns>An authentication token.</returns>
+        async Task<string> GetToken()
+        {
+            var client = new RestClient(_tokenUrl,
```

Comment:
Why is this file included? How is the OAuthAuthenticator used with ShipEngine? Why does it use RestClient instead of HttpClient?

Reply:
It's not necessary, but it's part of the default set of templates.  After #98 gets approved, I'm going to remove this and all other unmodified templates.